lib/parse-url.js

```diff
@@ -0,0 +1,33 @@
+import {parseInput as parseEmbeds} from 'embeds';
+import objectAssign from 'object-assign';
+import isUrl from 'is-url';
+import isImageUrl from 'is-image-url';
+
+export default (elm) => {
+  if (!elm) {
+    return null;
+  }
+
+  const content = elm.textContent || '';
+  if (!isUrl(content)) {
+    return null;
+  }
+
+  if (isImageUrl(content)) {
+    return {
+      type: 'embed',
+      embedType: 'image',
+      src: content
+    };
```

Comment:
also isImageUrl can make sync http calls - so a bit sceptical to that https://github.com/wzbg/is-image-url/blob/master/index.js#L22